### DISCUSSION TITLE
Function Argument Validation vs. parse

### DISCUSSION THREAD

**Discussion by Richard-Ott** [wschwanghart/topotoolbox#44] [General]
I though it might be worth thinking about switching from the input parser to the function argument validation. Seems like it would streamline TT functions a bit. 

https://nl.mathworks.com/help/matlab/matlab_prog/function-argument-validation-1.html#mw_b1fb2665-7be4-4fa5-a076-f82ad44304f4

**Comment by wkearn** [comment]
This is one of the changes that we intend to make on the way to v3. There's an open issue at https://github.com/TopoToolbox/topotoolbox/issues/6.

**Reply by wschwanghart** [reply]
Most of the functions in TT3 now have the argument block. There is only one drawback of this structure. If you need some computations to calculate the default values, then the argument block is somewhat restrictive as it needs to be the first line of code after the function definition. Still, the arguments block is great as it supports autocompletion when writing code.
